blackjax/mcmc/hmc.py

```diff
@@ -41,6 +41,11 @@ class HMCState(NamedTuple):
     logdensity_grad: ArrayTree
 
 
+class DynamicHMCState(NamedTuple):
+    hmc_state: HMCState
+    random_generator_arg: Array
```

Comment:
some comment here (mostly to highlight why we type it more general than a PRNGKey)